terraform/modules/spack/gitlab_webhooks.tf

```diff
@@ -16,3 +17,56 @@ resource "gitlab_project_hook" "job_webhook" {
 
 // TODO: Once https://gitlab.com/gitlab-org/terraform-provider-gitlab/-/issues/1350 is resolved the
 // gitlab_application_settings resource should be used to whitelist the domains in job_webhooks.
+
+
+data "gitlab_user" "spackbot" {
+  username = "spackbot"
+}
+
+resource "gitlab_personal_access_token" "webhook_handler" {
+  user_id = data.gitlab_user.spackbot.id
+  name    = "Webhook handler token"
+  # TODO: How to deal with this expiring
+  expires_at = "2024-12-03"
+
+  scopes = ["read_api", "read_repository"]
+}
+
+resource "random_password" "webhook_handler" {
+  length  = 64
+  special = false
+}
+
+# Note: the /1 is important to ensure that the broker for the webhook handler isn't using the same
+# database as the broker for spackbot.
+resource "kubectl_manifest" "webhook_secrets" {
+  yaml_body = <<-YAML
+     apiVersion: v1
+     kind: Secret
+     metadata:
+       name: webhook-secrets
+       namespace: custom
+     data:
+       gitlab-endpoint: ${base64encode("${var.gitlab_url}")}
+       gitlab-token: ${base64encode("${gitlab_personal_access_token.webhook_handler.token}")}
+       sentry-dsn: ${base64encode("${data.sentry_key.webhook_handler.dsn_public}")}
+       secret-key: ${base64encode("${random_password.webhook_handler.result}")}
+       celery-broker-url: ${base64encode("redis://${aws_elasticache_replication_group.pr_binary_graduation_task_queue.primary_endpoint_address}:6379/1")}
```

Comment:
It feels a bit weird that the gitlab webhook processor would rely on this specific redis cache. Would it not make sense to create a new (small) cache instance just for this webhook processor? Or to just deploy something like rabbitmq as part of the two existing deployments (if not its own deployment)?

My main concern is that these two seemingly unrelated parts of the infrastructure could end up affecting one another in some way.

Reply:
Yeah I discussed this with @mvandenburgh before since it seems a bit awkward. I think we're on the same page that we want to consolidate infrastructure so using a differently namespaced database `/1` on the same instance of redis makes sense. I'll watch the redis utilization after merge but we're almost comically overprovisioned at this point.

Reply:
FWIW, I pushed a change to explicitly listen on the celery queue and double checked that the spackbot queue uses database 0.